### PR TITLE
CI: Add `Mark issue duplicate` job

### PR DESCRIPTION
**Pull request by xrkffgg**
## Description

Add a workflow help you automatically manage issues.

ref: https://github.com/prettier/prettier/issues/9601 https://github.com/prettier/prettier/issues/10036

When you enter "Duplicate of #xxx",  the GitHub Actions will help you add the label `type:duplicate` and close the issue.

At the same time it supports concise commands, such as "/d", I don't know if you need it.

More details can see https://github.com/actions-cool/issues-helper/blob/main/README.en-US.md#mark-duplicate

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Could be useful.

**Comment by xrkffgg** [comment]
https://github.com/actions-cool/issues-helper also has many functions.

You can describe the functions you need when managing issues, and I will help you implement them.

**Comment by fisker** [comment]
I haven't check the codebase.

I have a silly question, when someone DON'T have commit access comments, will this action close issue?

**Comment by xrkffgg** [comment]
NO, when comment not "Duplicate of #xxx", it will not do anything.

**Comment by fisker** [comment]
That's not I'm asking.

**Comment by fisker** [comment]
When someone DON'T have commit access send comment "Duplicate of #xxx".

**Comment by xrkffgg** [comment]
> When someone DON'T have commit access send comment "Duplicate of #xxx".

Yes, this issue will be closed.

**Comment by fisker** [comment]
I don't like it.

File: .github/workflows/issue-mark-duplicate.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  mark-duplicate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: mark-duplicate
+        uses: actions-cool/issues-helper@1.x
+        with:
+          actions: 'mark-duplicate'
+          token: ${{ secrets.GITHUB_TOKEN }}
+          duplicate-labels: 'type:duplicate'
+          close-issue: true
```
Comment (xrkffgg) [inline .github/workflows/issue-mark-duplicate.yml:17]:
```suggestion

```

If you don't need to close the issue option, I can remove this.

Comment (fisker) [reply .github/workflows/issue-mark-duplicate.yml:17]: No, I want close it and add label, ONLY if the one have "commit access" send comment.

Comment (xrkffgg) [reply .github/workflows/issue-mark-duplicate.yml:17]: OK, let me update code to realize it.

**Comment by xrkffgg** [comment]
I added a parameter to control the permissions of the person who can do the operation.

At the same time, I have carried out verification in my project. See https://github.com/actions-cool/test-ci/issues/1

---

image

---
### Description

- `allow-permissions`: When you do not input, there is no limit. Anyone comment will trigger. Optional values are `admin`, `write`, `read`, `none`
  - If the team member sets the read permission, it is read
  - If the external Collaborator is set to read permission, it is read
  - Ordinary users have read permission

As you see the collaborator who has read right can't trigger the action.

**Comment by xrkffgg** [comment]
The github api i uesd https://docs.github.com/en/rest/reference/repos#get-repository-permissions-for-a-user

**Comment by xrkffgg** [comment]
Has updated. By default, required `write` above permission.

**Review by fisker — APPROVED** [review]
LGTM, thanks.

File: .github/workflows/mark-issue-duplicate.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+on:
+  issue_comment:
+    types:
+      - created
+
+jobs:
+  mark-duplicate:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions-cool/issues-helper@v2
```
Comment (xrkffgg) [inline .github/workflows/mark-issue-duplicate.yml:12]:
```suggestion
      - uses: actions-cool/issues-helper@v2.0.0
```

**Review by thorn0 — APPROVED** [review]

**Comment by fisker** [comment]
@xrkffgg  What happens if I comment duplicate of #xxx on  a PR? Close PR?

**Comment by xrkffgg** [comment]
> What will happen if I comment `duplicate of #xxx` on a PR? 

Nothing, I checked `Duplicate of`. And if comment body include `?`, also not trigger.

**Comment by xrkffgg** [comment]
If comment `duplicate of`, it will no noting. I check `Duplicate of`

**Comment by xrkffgg** [comment]
Of course, you can say what you want at the same time, and I will improve it.

**Comment by thorn0** [comment]
@xrkffgg I wonder what happened here:
https://github.com/prettier/prettier/runs/1814759559?check_suite_focus=true

**Comment by xrkffgg** [comment]
let me check

**Comment by xrkffgg** [comment]
It's strange, I am normal here. https://github.com/actions-cool/test-ci/issues/5

**Comment by fisker** [comment]
Is it some api reach limitation?

**Comment by xrkffgg** [comment]
```
(node:2391) UnhandledPromiseRejectionWarning: HttpError: Resource not accessible by integration
    at /home/runner/work/_actions/actions-cool/issues-helper/v2.0.0/dist/index.js:3467:23
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async doMarkDuplicate (/home/runner/work/_actions/actions-cool/issues-helper/v2.0.0/dist/index.js:8032:17)
    at async choseActions (/home/runner/work/_actions/actions-cool/issues-helper/v2.0.0/dist/index.js:8462:11)
```



the reason is `getCollaboratorPermissionLevel` api error.

**Comment by xrkffgg** [comment]
https://www.githubstatus.com/

**Comment by xrkffgg** [comment]
image

**Comment by xrkffgg** [comment]
> Is it some api request reach limitation?

Public repositorie no limit

**Comment by fisker** [comment]
Do you think we should add a condition here https://github.com/prettier/prettier/blob/e024ec2d3f886cdbf35c2b5cf87877009e4d17a8/.github/workflows/mark-issue-duplicate.yml#L10 ? like https://github.com/sosukesuzuki/prettier-regression-testing/blob/3e5b8638f25396ebdb3d9632b44e2be4d3e6ea86/.github/workflows/main.yml#L6, So it will skip none related comments.

**Comment by xrkffgg** [comment]
Yey it's more perfeact

`if: startsWith(github.event.comment.body, 'Duplicate')`

**Comment by xrkffgg** [comment]
image

https://www.githubstatus.com/

Maybe this is the reason. @thorn0

**Comment by xrkffgg** [comment]
test

**Comment by xrkffgg** [comment]
test

**Comment by thorn0** [comment]
Thanks!

**Comment by xrkffgg** [comment]
I need to check again, but it is still very strange.

**Comment by xrkffgg** [comment]
I wonder if you have modified this. Because the default `github-bot` is used, it should have permission to make API call queries.

I saw this https://github.com/prettier/prettier/issues/10202 is normal yestday. My local test is also normal.

https://docs.github.com/en/github/setting-up-and-managing-your-enterprise/enforcing-github-actions-policies-in-your-enterprise-account

**Comment by thorn0** [comment]
> I wonder if you have modified this.

Modified what exactly?

**Comment by xrkffgg** [comment]
GitHub Actions policies like [docs](https://docs.github.com/en/github/setting-up-and-managing-your-enterprise/enforcing-github-actions-policies-in-your-enterprise-account)

Now it is in settings/actions

**Comment by thorn0** [comment]
No, nobody touched that

**Comment by xrkffgg** [comment]
Because under normal circumstances, using `github-bot` to request the API will not report an error, so I did not display the exception description of the api request.

But when I look at the current Actions log, error still. I guess it is a permission problem, but I need to output the log to confirm.

**Comment by xrkffgg** [comment]
I think I found the reason

https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits

**Comment by xrkffgg** [comment]
> > Is it some api request reach limitation?
> 
> Public repositorie no limit

Api has limit . See https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting

GitHub Actions frequency no limit.

I will improve it and reduce unnecessary requests.

**Pull request merged**